ompi/mca/mtl/ofi/mtl_ofi.h

```diff
@@ -110,10 +110,15 @@ ompi_mtl_ofi_progress(void)
                 abort();
             }
         } else {
-            /**
-             * The CQ is empty. Return.
-             */
-            break;
+            /* CQ is empty, return */
+            if (ret == -FI_EAGAIN) {
+                break;
+            } else {
+                opal_output_verbose(1, ompi_mtl_base_framework.framework_output,
+                                    "%s:%d: fi_cq_read failed: %s(%zd)\n",
+                                    __FILE__, __LINE__, fi_strerror(-ret), ret);
+                abort();
```

Comment:
Doesn't `abort()` drop a corefile?  I don't think you want to do that -- especially at scale.

Reply:
Right. Will fix this. But this is again prevalent in other areas of the code as well (ompi_mtl_ofi_finalize()). Shall I fix these as well?

Reply:
Yes -- dropping corefiles is quite anti-social.  Especially if N Open MPI procs do it all at once -- it'll hammer networked filesystems, and can quickly cause you to run out of disk space.

Reply:
@jsquyres  Generic question- If we reach this point, it is to be interpreted as a fatal error and we need to stop execution right away. Is there a way to hint `opal_progress()` (which is higher up call stack) to "abort"? `opal_progress()` itself is void function and doesn't take any parameters. So not sure how to do this.

Reply:
I don't have the cycles to look at this immediately, but in the OB1 PML, there's a callback function for making the process fail (i.e., the BTL can invoke a callback back into OB1).  I don't know if the same mechanism exists in the CM MTL, but if it does, that's probably what you should use.

Reply:
I looked at this and found the error callback function. But I didn't find a similar mechanism in CM PML. From looking at `opal_btl_usnic_exit()`, I see that it searches for a valid callback and if present, calls it. If not, we print message using stderr and simply `exit(1)` out of here. This seems a nice approach to stop execution without dumping a corefile. I can follow the same approach here. 

Thoughts?

Reply:
👍

Reply:
Fixed this and used `opal_show_help()` for fail paths of some basic OFI calls.